### ISSUE TITLE
resume parser 

### ISSUE THREAD

**Issue by rishi23root** [rishi23root/resume-curator#31]
**Aim**: Take input of  resume pdf or doc file parse it, and return [json resume](https://jsonresume.org/schema/) format in return with at least 70-80% accuracy (_more is always better_)

### For now, we expect it just to be a function, and the PDF file is coming from a local storage (for testing use output folder files)
- [ ] load the file and read the file 
- [ ] extract just text from the file
- [ ] load a pre-trained model to parse the data and return in JSON format (expecting all the fields to be present, all unprocessed fields are left empty)
- [ ] if not (3) then fill the JSON with the remaining missing fields
- [ ] return the data 

## In the future, this function will be linked to an API endpoint
- the file will be streamed and stored in the temp folder and read in realtime 
- need to create an API endpoint 

## For model Development
- use a folder named `parser` in the root of the project  
- will be used to store parser code and training data and out model 
- and all the helper functions related to parsing
- and also the wrapper for the build model and its usable instance provider

## Constraints
- should be in Python
- regex base parsers are just reliable but can be used with the combination of some nlp or nn model

## Resources to use (but not seem promising)
- [Google search top results 2nd link](https://resume-parser.vercel.app/) -  using 3rd party API which is paid for full access
- [hugging face filted](https://huggingface.co/models?sort=trending&search=resume+parser) - seems promising 
- [open resume](https://www.open-resume.com/) - regex base and not python 
- [keggle repos](https://www.kaggle.com/search?q=resume+parser+notebookLanguage%3APython) - doesn't seem to meet the requirements, maybe need more drilling
- [python lib](https://pypi.org/project/resume-parser/) - doesn't seem to meet requirements, but still best one so far
- The last option custom train model, which can use - https://spacy.io/
